automatic/nexus-repository/tools/chocolateyinstall.ps1

```diff
@@ -0,0 +1,144 @@
+$ErrorActionPreference = 'Stop'
+$ToolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+Import-Module (Join-Path $ToolsDir 'supportingfunctions.psm1')
+
+if (Get-OSArchitectureWidth 32) {
+    throw "Sonatype Nexus Repository 3.0 and greater only supports 64-bit Windows."
+}
+
+$Version = '[[VERSION]]'
+$NexusVersionedFolder = "nexus-$Version"
+$TargetFolder = "$env:ProgramData\nexus"
+$ExtractFolder = "$env:TEMP\NexusExtract"
+$TargetDataFolder = "$env:ProgramData\sonatype-work"
+$NexusConfigFile = "$TargetDataFolder\nexus3\etc\nexus.properties"
+$ServiceName = 'nexus'
+
+# Handle Package Parameters
+$pp = Get-PackageParameters
+
+$Hostname = if ($pp.ContainsKey("Fqdn")) {
+    $pp["Fqdn"]
+} else {
+    "localhost"
+}
+
+$NexusPort = if ($pp.ContainsKey("Port")) {
+    $pp["Port"]
+    Write-Host "/Port was used, Nexus will listen on port $($NexusPort)."
+} else {
+    "8081"
+}
+
+if (Test-Path "$env:ProgramFiles\nexus\bin") {
+    throw "Previous version of Nexus 3 installed by setup.exe is present, please uninstall before running this package."
+}
+
+if ((Get-Service $ServiceName -ErrorAction SilentlyContinue)) {
+    Write-Warning "Nexus web app is already present, shutting it down so that we can upgrade it."
+    Get-Service $ServiceName | Stop-Service -Force
+}
+
+if ($pp.ContainsKey("BackupSslConfig")) {
+    if ($pp.ContainsKey("BackupLocation")) {
+        Backup-NexusSSL -BackupLocation $pp["BackupLocation"]
+    } else {
+        Backup-NexusSSL
+    }
+}
+
+# Extract the Nexus program files, and overwrite any previous program files
+if (Test-Path "$ExtractFolder") {
+    Remove-Item "$ExtractFolder" -Recurse -Force
+}
+
+$PackageArgs = @{
+    packageName    = $env:ChocolateyPackageName
+    unzipLocation  = $ExtractFolder
+    url64          = '[[URL]]'
+    checksum64     = '[[CHECKSUM]]'
+    checksumType64 = 'SHA256'
+}
+
+Install-ChocolateyZipPackage @PackageArgs
+
+Write-Host "Copying files to '$TargetFolder' with overwrite"
+if (Test-Path "$TargetFolder") {
+    Copy-Item "$ExtractFolder\$nexusversionedfolder\*" "$TargetFolder" -Force -Recurse
+} else {
+    Copy-Item "$ExtractFolder\$nexusversionedfolder" "$TargetFolder" -Force -Recurse
+}
+
+# Create the Nexus data directory, if it doesn't exist
+if (!(Test-Path "$TargetDataFolder")) {
+    Move-Item "$extractfolder\sonatype-work" "$TargetDataFolder"
+} else {
+    Write-Warning "`"$TargetDataFolder`" already exists, not overwriting, residual data from previous installs will not be reset."
+}
+
+Remove-Item "$ExtractFolder" -Force -Recurse
+
+# Install the Nexus service
+$processArgs = @{
+    ExeToRun       = "$TargetFolder\bin\nexus.exe"
+    Statements     = "/install $servicename"
+    validExitCodes = @(0)
+}
+
+$null = Start-ChocolateyProcessAsAdmin @processArgs
```

Comment:
Any reason you need to use `Start-ChocolateyProcessAsAdmin` instead of using `Install-ChocololateyInstallPackage`?

Reply:
To my knowledge, this file isn't an installer - it's just an argument running on the unpacked binary that triggers the addition of a service. That is, to be fair, a bit of an odd argument now I write it out, but to my mind it makes sense that this is just starting a process rather than installing a thing - the install of the software happens with the `Install-ChocolateyZipPackage`, above.